### PR TITLE
refactor to Promise. Closes #11

### LINKED ISSUE #11: Using promises
Right now the plan is to follow callback pattern. We might want to provide a way to return promise from each method or even make them promise-based only.

cc @nikgraf @eahefnawy @pmuens

### PR DESCRIPTION
**Pull request by mthenw**

### CODE REVIEW

File: stdlib.js [hunk 1]
```diff
@@ -5,52 +5,47 @@ class Stdlib {
...
+            reject(new Error(`Calling function failed: ${err.message}`));
+          } else {
+            let payload;
+            try {
+              payload = JSON.parse(response.Payload);
+
+              if (response.FunctionError) {
+                reject(new Error(`Calling function failed: ${payload.errorMessage}`));
+              } else {
+                resolve(payload);
```
Comment (nikgraf) [inline stdlib.js]: What if the response is a binary?

Comment (mthenw) [inline stdlib.js]: Accoring to spec `response.Payload` is always JSON parsable

> It is the JSON representation of the object returned by the Lambda function. This is present only if the invocation type is RequestResponse.

Comment (nikgraf) [inline stdlib.js]: ahhhh, my mistake. It was a API Gateway announcement!

**Comment by nikgraf** [comment]
Good to be merged!

**Pull request merged**